crates/server/src/graphql/modules/post/mutation/post_create.rs

```diff
@@ -0,0 +1,51 @@
+use async_graphql::{Context, Result};
+use gabble::post::service::CreatePostDto;
+use pxid::Pxid;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    context::SharedContext,
+    graphql::modules::post::types::{Post, PostError, PostErrorCode},
+};
+
+#[derive(Debug, Default)]
+pub struct PostCreateInput {
+    pub author_id: String,
+    pub parent_id: Option<Pxid>,
+    pub head: bool,
```

Comment:
I think head is an internal value and must be handled by the service and not the user.